### PR TITLE
Add Java 17 images

### PR DESCRIPTION
**Pull request by onobc**
Contributes to https://github.com/Azure/azure-functions-java-worker/issues/548

For [Spring Cloud Function Azure](https://github.com/spring-cloud/spring-cloud-function/tree/main/spring-cloud-function-samples/function-sample-azure) we need to run Java 17 on Azure as the latest version of Spring Cloud Function requires it. However, the Azure Java functions runtime [does not yet support Java 17](https://github.com/Azure/azure-functions-java-worker/issues/5480).

To get around this limitation we deploy to Azure in a custom Docker container as described [here](https://docs.microsoft.com/en-us/azure/azure-functions/functions-create-function-linux-custom-image?tabs=in-process%2Cbash%2Cazure-cli&pivots=programming-language-java). One problem, no Java 17 images in this repo to base our image from. I created the Java17 images from this repo locally and it works well. I figured I would attempt to contribute this back. I am sure there are other hurdles besides this PR to "turn on Java 17 for the world" but here we go anyways. I hope it helps.

> ℹ️ I am not super familiar w/ this repo pipeline and tests so I am going to submit this PR and watch what the CI does. I am happy to update/add tests for this. Any guidance is appreciated as well. Thanks.

### PR information

- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/development-docs/cleaning-up-commits.md).
- [x] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made and CI is passing.

### CODE REVIEW

File: host/4/bullseye/amd64/java/java17/java17-composite.template [hunk 1]
```diff
@@ -0,0 +1,41 @@
+ARG JAVA_VERSION=17.0.2.8.1
+ARG JAVA_HOME=/usr/lib/jvm/msft-17-x64
+FROM mcr.microsoft.com/dotnet/runtime-deps:6.0.0
+ARG HOST_VERSION
+ARG JAVA_VERSION
+ARG JAVA_HOME
+
```
Comment (kaibocai) [inline host/4/bullseye/amd64/java/java17/java17-composite.template:7]: can we remove extra space line here. thanks.

**Comment by kaibocai** [comment]
Hi @onobc, thanks very much for your contribution! Left some minor comment. Thanks again!

File: host/4/core-tools-publish.yml [hunk 2]
```diff
@@ -551,6 +563,45 @@ stages:
...
+          docker push $TARGET_REGISTRY/java:$(TargetVersion)-java17-core-tools
+
+          docker push $TARGET_REGISTRY/java:$(MajorVersion)-java17-core-tools
+          
+          docker system prune -a -f
+        displayName: push java17 images
+        continueOnError: false
+        condition: ne(variables['SkipProduction'], 'true')
+
+      - bash: |
```
Comment (kaibocai) [inline host/4/core-tools-publish.yml:594]: why we push node images here?

**Comment by kaibocai** [comment]
Please let me know if I can modify your pr directly. Thanks.

**Comment by onobc** [comment]
Hi @kaibocai , yes feel free to modify the PR in any way you need.

**Review by pragnagopa — CHANGES_REQUESTED** [review]
Let’s discuss offline on plans to support Java 17

@kaibocai please work with @CooperLink to ensure layers are shared - this is after we make a decision on merging this PR

**Comment by kaibocai** [comment]
close this one with a new pr https://github.com/Azure/azure-functions-docker/pull/714

**Pull request closed without merging**